src/hammock.js

```diff
@@ -0,0 +1,209 @@
+// @flow
+/* global fetch */
+import { createAction } from 'redux-actions'
+import R from 'ramda'
+import _ from 'lodash'
+
+import type { Action, ActionType, Dispatcher } from './reduxTypes'
+import type { Endpoint } from './restTypes'
+// import { fetchJSONWithCSRF } from './api';
+import {
+  FETCH_PROCESSING,
+  FETCH_SUCCESS,
+  FETCH_FAILURE,
+  GET,
+  PATCH,
+  POST,
+  INITIAL_STATE
+} from './constants'
+import { withUsername, updateStateByUsername } from './util'
+
+const actionize = R.compose(R.toUpper, R.join('_'), R.map(_.snakeCase))
+
+export const requestActionType = (...xs: string[]) => `REQUEST_${actionize(xs)}`
+
+export const successActionType = (...xs: string[]) => `RECEIVE_${actionize(xs)}_SUCCESS`
+
+export const failureActionType = (...xs: string[]) => `RECEIVE_${actionize(xs)}_FAILURE`
+
+export const clearActionType = (...xs: string[]) => `CLEAR_${actionize(xs)}`
+
+const defaultRESTPrefixes = { GET, PATCH, POST }
+
+const getPrefixForEndpoint = (verb, endpoint) => (
+  R.propOr(defaultRESTPrefixes[verb], `${R.toLower(verb)}Prefix`, endpoint)
+)
+
+const getUrl = (endpoint: Endpoint, verb: string) => (
+  endpoint[`${R.toLower(verb)}Url`]
+)
+
+const simpleOptions = () => ({})
+
+const getMakeOptions = (endpoint: Endpoint, verb: string) => (
+  endpoint[`${R.toLower(verb)}Options`] || simpleOptions
+)
+
+export function makeFetchFunc (endpoint: Endpoint, verb: string): (...args: any) => Promise<*> {
+  let url = getUrl(endpoint, verb)
+  let options = getMakeOptions(endpoint, verb)
+  return (...args) => {
+    return fetch(_.isFunction(url) ? url(...args) : url, options(...args))
+  }
+};
```

Comment:
Not important but I'm wondering if the linter rules can be changed to detect this semicolon and the next one